src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,164 @@
+ */
+public class ImportCommand extends Command {
+
+    public static final String COMMAND_WORD = "import";
+    public static final String MESSAGE_SUCCESS = "Contacts imported successfully from: %1$s";
+    public static final String MESSAGE_FAILURE = "Failed to import contacts from: %1$s";
+    public static final String MESSAGE_INVALID_CATEGORY = "Invalid category in CSV. Category must be either 'student' "
+            + "or 'company'";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports file in comma-separated value(CSV) format.\n"
+            + "Example: " + COMMAND_WORD + " /path/to/file.csv";
+    private final String filePath;
+
+    /**
+     * Creates an ImportCommand to import contacts from the specified CSV file path.
+     *
+     * @param filePath The path of the CSV file to import contacts from.
+     */
+    public ImportCommand(String filePath) {
+        requireNonNull(filePath);
+        this.filePath = filePath;
+    }
+
+    /**
+     * Executes the ImportCommand, reading contacts from the specified CSV file
+     * and adding them to the address book.
+     *
+     * @param model The model in which contacts are to be added.
+     * @return A CommandResult indicating success or failure of the import operation.
+     * @throws CommandException If an error occurs during file reading or processing.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        int successCount = 0;
+
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            String line = br.readLine(); // Skip the header row
+
+            while ((line = br.readLine()) != null) {
+                String[] values = line.split(","); // Split CSV by commas
+
+                if (values.length < 6) {
+                    throw new CommandException("Invalid CSV format");
+                }
+
+                Name name = new Name(values[0].trim());
+                String category = values[1].trim();
+                Phone phone = new Phone(values[3].trim());
+                Email email = new Email(values[4].trim());
+                Address address = new Address(values[5].trim());
+                Set<Tag> tagSet = parseTags(values.length > 6 ? values[6].trim() : "");
+
+                switch (category.toLowerCase()) {
+                case "student":
+                    String studentID = values[2].trim();
+                    if (!studentID.isEmpty()) {
+                        Student student = new Student(name, new StudentID(studentID), phone, email, address, tagSet);
+                        if (!model.hasPerson(student)) {
+                            new AddStudentCommand(student).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Student ID for student category");
```

Comment:
Perhaps can add a test where the category is `student` and the `studentId` is missing to check if this exception is thrown